src/edu/csus/ecs/pc2/clics/API202003/EventFeedJSON.java

```diff
@@ -331,24 +314,33 @@ public String getTeamMemberJSON(IInternalContest contest) {
 
         StringBuilder stringBuilder = new StringBuilder();
 
-        Account[] accounts = getTeamAccounts(contest);
+        Account[] accounts = EventFeedJSON.getTeamAccounts(contest);
 
         for (Account account : accounts) {
             String[] names = account.getMemberNames();
 
             if (names.length > 0) {
                 for (String teamMemberName : names) {
-                    appendJSONEvent(stringBuilder, TEAM_MEMBERS_KEY, ++eventIdSequence, EventFeedOperation.CREATE, getTeamMemberJSON(contest, account, teamMemberName));
-                    stringBuilder.append(NL);
+                    String teamMemberJson = getTeamMemberJSON(contest, account, teamMemberName);
+                    if(teamMemberJson != null) {
+                        appendJSONEvent(stringBuilder, TEAM_MEMBERS_KEY, ++eventIdSequence, EventFeedOperation.CREATE, teamMemberJson);
+                        stringBuilder.append(NL);
+                    }
                 }
             }
         }
 
         return stringBuilder.toString();
     }
 
-    protected String getTeamMemberJSON(IInternalContest contest, Account account, String teamMemberName) {
-        return teamMemberJSON.createJSON(contest, account, teamMemberName);
+    public String getTeamMemberJSON(IInternalContest contest, Account account, String teamMemberName) {
+        return(null);
+        // TODO: Fix in IJSONTool
+        // The class TeamMemberJSON which supposedly generates JSON for a team member is just plain wrong.
+        // It gives all members of a team the same "id" for starters.  Also, for some reason, it only added team
+        // id's if it had an external CMS team id, even though it was adding the pc2 team's client ID.
+        // Basically it was just plain wrong.
```

Comment:
I wonder if the issue described in this comment doesn't rise to the level of "this is so important to fix that we shouldn't just be "returning null" when this method is called, but rather we should be doing (something like) `throw new UnsupportedOperationException("...msg...")` so that we remember that we NEED to fix this?  Otherwise, I worry that this will get lost in the (probably literally _thousands_) of "TODOs" sprinkled throughout the code.  What's the ramification of `return(null)`?  Doesn't this pose the likelihood of some piece of "client code" failing because, well, a "team member" shouldn't ever be null?   Or do we expect client code to _always_ check for null return values?

Reply:
Anything that uses `getTeamMemberJSON()` does check for null.  I don't think it was ever an issue because we never add team members to a contest (there isn't a mechanism to do it from the snake yaml reader).  We can do it from the GUI using the "auto registration" stuff, which we have no way of using at the moment anyway.  (Recall we discussed this too, that pc2 does have built-in auto-team registration, but it's never been tested).

Reply:
Ok, I guess this falls into "lets not worry about it until we actually start working in earnest on something that would use it, like 'AutoRegistration'".